### PR TITLE
docs: Clarify manual graceful shutdown in dev server

### PR DESCRIPTION
**Pull request by manovotny**
Replaces #50481

### CODE REVIEW

File: docs/02-app/01-building-your-application/07-deploying/index.mdx [hunk 1]
```diff
@@ -138,10 +138,12 @@ Sometimes you might want to run some cleanup code on process signals like `SIGTE
 
 You can do that by setting the env variable `NEXT_MANUAL_SIG_HANDLE` to `true` and then register a handler for that signal inside your `_document.js` file. Please note that you need to register env variable directly in the system env variable, not in the `.env` file.
```
Comment (delbaoliveira) [inline docs/02-app/01-building-your-application/07-deploying/index.mdx:139]:
```suggestion
You can set the env variable `NEXT_MANUAL_SIG_HANDLE` to `true` and then register a handler for that signal inside your `_document.js` file. Please note that you need to register the env variable directly in the system env variable, not in the `.env` file.
```

**Review by delbaoliveira — DISMISSED** [review]

File: docs/02-app/01-building-your-application/07-deploying/index.mdx [hunk 2]
```diff
@@ -136,12 +136,14 @@ Next.js will automatically load the latest version of your application in the ba
 
 Sometimes you might want to run some cleanup code on process signals like `SIGTERM` or `SIGINT`.
 
-You can do that by setting the env variable `NEXT_MANUAL_SIG_HANDLE` to `true` and then register a handler for that signal inside your `_document.js` file. Please note that you need to register env variable directly in the system env variable, not in the `.env` file.
+You can set the env variable `NEXT_MANUAL_SIG_HANDLE` to `true` and then register a handler for that signal inside your `_document.js` file. Please note that you need to register the env variable directly in the system env variable, not in the `.env` file.
```
Comment (leerob) [inline docs/02-app/01-building-your-application/07-deploying/index.mdx:139]:
```suggestion
You can set the env variable `NEXT_MANUAL_SIG_HANDLE` to `true` and then register a handler for that signal inside your `_document.js` file. You will need to register the env variable directly in the `package.json` script, not in the `.env` file.
```

I wonder what "system env var" means here - package.json script?

File: docs/02-app/01-building-your-application/07-deploying/index.mdx [hunk 3]
```diff
@@ -136,12 +136,14 @@ Next.js will automatically load the latest version of your application in the ba
 
 Sometimes you might want to run some cleanup code on process signals like `SIGTERM` or `SIGINT`.
```
Comment (leerob) [inline docs/02-app/01-building-your-application/07-deploying/index.mdx:137]: When self-hosting, you might want to run code when the server shuts down on `SIGTERM` or `SIGINT` signals.

Comment (leerob) [inline docs/02-app/01-building-your-application/07-deploying/index.mdx:137]:
```suggestion
When self-hosting, you might want to run code when the server shuts down on `SIGTERM` or `SIGINT` signals.
```

**Review by leerob — APPROVED** [review]

**Pull request merged**